lib/backend/client/hostendpoint.go

```diff
@@ -0,0 +1,52 @@
+package client
+
+import (
+	. "github.com/projectcalico/libcalico/lib/backend/objects"
+)
+
+// HostEndpointInterface has methods to work with Service resources.
+type HostEndpointInterface interface {
+	List(hostname, name *string) ([]HostEndpoint, error)
+	Get(hostname, name string) (*HostEndpoint, error)
+	Create(hostname, name string, data *HostEndpoint) error
+	Update(hostname, name string, data *HostEndpoint) error
+	Delete(hostname, name string) error
+}
+
+// services implements ServicesNamespacer interface
+type hostEndpoints struct {
+	r  *Client
+}
+
+// newServices returns a services
+func newHostEndpoints(c *Client) *hostEndpoints {
+	return &hostEndpoints{c}
+}
+
+// List takes a Metadata, and returns the list of hot endpoints that match that Metadata
+// (wildcarding mising fields)
+func (h *hostEndpoints) List(hostname, name *string) ([]HostEndpoint, error) {
+	return nil, nil
+}
+
+// Get returns information about a particular host endpoint.
+func (h *hostEndpoints) Get(hostname, name string) (*HostEndpoint, error) {
+	return nil, nil
+}
+
+// Create creates a new host endpoint.
+func (h *hostEndpoints) Create(hostname, name string, data *HostEndpoint) error {
```

Comment:
Aren't we embedding the hostname and name in the HostEndpoint struct?  Why do we need to pass them in too?